google-cloud-storage/src/main/java/com/google/cloud/storage/GrpcStorageImpl.java

```diff
@@ -402,6 +403,33 @@ public Blob get(BlobId blob) {
     return get(blob, new BlobGetOption[0]);
   }
 
+  @Override
+  public Blob restore(BlobId blob, BlobRestoreOption... options) {
+    Opts<ObjectSourceOpt> unwrap = Opts.unwrap(options);
+    return internalObjectRestore(blob, unwrap);
+  }
+
+  private Blob internalObjectRestore(BlobId blobId, Opts<ObjectSourceOpt> opts) {
+    Opts<ObjectSourceOpt> finalOpts = opts.prepend(defaultOpts).prepend(ALL_BLOB_FIELDS);
+    GrpcCallContext grpcCallContext =
+        finalOpts.grpcMetadataMapper().apply(GrpcCallContext.createDefault());
+    RestoreObjectRequest.Builder builder =
+        RestoreObjectRequest.newBuilder()
+            .setBucket(bucketNameCodec.encode(blobId.getBucket()))
+            .setObject(blobId.getName());
+    ifNonNull(blobId.getGeneration(), builder::setGeneration);
```

Comment:
Is the API failure helpful when generation isn't supplied?

Reply:
Yeah, it says "you must specify a generation"